File: abc/abc162/abc162_b.cs
// https://atcoder.jp/contests/abc162/submissions/11818801

using System;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var N = long.Parse(Console.ReadLine());

            var result = 0L;
            for (int i = 1; i <= N; i++)
            {
                if (i % 3 > 0 && i % 5 > 0)
                {
                    result += i;
                }
            }
            Console.WriteLine(result);
        }
    }
}

File: abc/abc162/abc162_b.fs
// https://atcoder.jp/contests/abc162/submissions/11856931

let N = stdin.ReadLine() |> int64

let rec solver s n =
    match n with
    | i when i = N + 1L -> s
    | i when i % 3L > 0L && i % 5L > 0L -> solver (s + n) (n + 1L)
    | _ -> solver s (n + 1L)

solver 0L 0L |> stdout.WriteLine
